packages/editor/src/hooks/pattern-partial-syncing.js

```diff
@@ -64,10 +64,44 @@ const withPartialSyncingControls = createHigherOrderComponent(
 	}
 );
 
+/**
+ * Adds an editOriginalPattern prop which allows the block to switch between post and pattern
+ * editing modes.
+ *
+ * @param {Component} BlockEdit Original component.
+ *
+ * @return {Component} Wrapped component.
+ */
+const withPatternOnlyRenderMode = createHigherOrderComponent(
```

Comment:
What is this filter and why we need it?

Reply:
This was just a temporary way of previously getting the `editor` specific rendering mode call into the `core/block` block without making `editor` a dependency of `block-library` for the purposes of quickly trying PcC. Now we are moving to using the `onSelectPost` setting instead we will be deleting this temporary filter.